dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/datavalueset/DefaultDataValueSetService.java

```diff
@@ -935,15 +935,16 @@ private void saveDataValueUpdate( ImportContext context, ImportCount importCount
         }
         if ( !internalValue.isDeleted()
             && Objects.equals( existingValue.getValue(), internalValue.getValue() )
-            && Objects.equals( existingValue.getComment(), internalValue.getComment() ) )
+            && Objects.equals( existingValue.getComment(), internalValue.getComment() )
+            && existingValue.isFollowup() == internalValue.isFollowup() )
```

Comment:
Good catch here, agreed. Now should we not due == checks on object Booleans and instead use equals (taking null into account)? It will work in practice but I recall that regression around boolean system settings and Redis.

Reply:
These are `boolean`s